src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -76,6 +88,11 @@ internal enum ScopeType
             Padding = 2,
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating how to escape the string when writing JSON string.
+        /// </summary>
+        internal ODataStringEscapeOption StringEscapeOption { get; set; }

```

Comment:
This field is not supposed to be set after the JsonWriter instance is instantiated. We should remove the setter method and make it read-only, similar to other fields such as isIeee754Compatible.

Reply:
See the new added comments in "ODataJsonLightOutputContext.cs"

Reply:
link?

Reply:
```C#
// We can't change the public interface "IJsonWriterFactory" to avoid breaking changes. 
// So, use the following work around. 
           JsonWriter defaultJsonWriter = jsonWriter as JsonWriter; 
            if (defaultJsonWriter != null) 
            { 
                defaultJsonWriter.StringEscapeOption = stringEscapeOption; 
            } 
```

Reply:
IMHO, this is a bit hacky.

Reply:
Maybe. Do you have a better solution?

Reply:
please check with @mikepizzo

Reply:
What if we let the user create the JSON Writer (and specify the appropriate escape option) and then give it to us through DI?  Then we wouldn't have to add the property to ODataMessageWriterSettings.

---
In reply to: [215032187](https://github.com/OData/odata.net/pull/1242#discussion_r215032187) [](ancestors = 215032187)

Reply:
Added comments to previous issue.

---
In reply to: [215051592](https://github.com/OData/odata.net/pull/1242#discussion_r215051592) [](ancestors = 215051592)

Reply:
Added comments to previous issue.

---
In reply to: [215057928](https://github.com/OData/odata.net/pull/1242#discussion_r215057928) [](ancestors = 215057928)

Reply:
This is internal, so I don't mind making it r/w and having it internally set. I don't like having a messagewriter setting that only applies in this case, though.  See my comments below on an potential alternative design.

---
In reply to: [215038804](https://github.com/OData/odata.net/pull/1242#discussion_r215038804) [](ancestors = 215038804)